modules/core/infrastructure/persistence/authlog_repository.go

```diff
@@ -14,91 +23,124 @@ func NewAuthLogRepository() authlog.Repository {
 }
 
 func (g *GormAuthLogRepository) GetPaginated(
-	ctx context.Context,
-	limit,
-	offset int,
-	sortBy []string,
+	ctx context.Context, params *authlog.FindParams,
 ) ([]*authlog.AuthenticationLog, error) {
-	tx, ok := composables.UseTx(ctx)
-	if !ok {
-		return nil, composables.ErrNoTx
+	pool, err := composables.UsePool(ctx)
+	if err != nil {
+		return nil, err
+	}
+	where, args := []string{"1 = 1"}, []interface{}{}
+	if params.ID != 0 {
+		where, args = append(where, fmt.Sprintf("id = ANY($%d)", len(args)+1)), append(args, params.ID)
 	}
-	q := tx.Limit(limit).Offset(offset)
-	q, err := helpers.ApplySort(q, sortBy)
+
+	if params.UserID != 0 {
+		where, args = append(where, fmt.Sprintf("user_id = $%d", len(args)+1)), append(args, params.UserID)
+	}
+
+	rows, err := pool.Query(ctx, `
+		SELECT id, user_id, ip, user_agent, created_at
+		FROM authentication_logs
+		WHERE `+strings.Join(where, " AND ")+`
+		ORDER BY id DESC
+		`+repo.FormatLimitOffset(params.Limit, params.Offset),
+		args...,
+	)
 	if err != nil {
 		return nil, err
 	}
-	var entities []*authlog.AuthenticationLog
-	if err := q.Find(&entities).Error; err != nil {
+	defer rows.Close()
+	logs := make([]*authlog.AuthenticationLog, 0)
+	for rows.Next() {
+		var log models.AuthenticationLog
+		if err := rows.Scan(
+			&log.ID,
+			&log.UserID,
+			&log.IP,
+			&log.UserAgent,
+			&log.CreatedAt,
+		); err != nil {
+			return nil, err
+		}
+		logs = append(logs, toDomainAuthenticationLog(&log))
+	}
+	if err := rows.Err(); err != nil {
 		return nil, err
 	}
-	return entities, nil
+	return logs, nil
 }
 
 func (g *GormAuthLogRepository) Count(ctx context.Context) (int64, error) {
-	tx, ok := composables.UseTx(ctx)
-	if !ok {
-		return 0, composables.ErrNoTx
+	pool, err := composables.UsePool(ctx)
```

Comment:
You should use `UsePoolTx`. In case `Count` gets called inside a transaction wrapped request, the user will expect changes done inside transaction to affect the call to `Count`